.travis.yml

```
@@ -1,24 +1,54 @@
 language: minimal
-sudo: required
-env:
-  matrix:
-  - OS_TYPE=centos OS_VERSION=6 BUILD_ENV=uw_build
-  - OS_TYPE=centos OS_VERSION=7 BUILD_ENV=uw_build
-  - OS_TYPE=centos OS_VERSION=6 BUILD_ENV=osg
-  - OS_TYPE=centos OS_VERSION=7 BUILD_ENV=osg
-
-services:
-  - docker
-  
-before_install:
-  - sudo apt-get update
-  - echo 'DOCKER_OPTS="-H tcp://127.0.0.1:2375 -H unix:///var/run/docker.sock -s devicemapper"' | sudo tee /etc/default/docker > /dev/null
-  - sudo service docker restart
-  - sleep 5
-  - sudo docker pull centos:centos${OS_VERSION}
-
-  
-script:
- # Run tests in Container
-- tests/setup_tests.sh
+jobs:
+  include:
+    - &run_tests
+      # Template; subsequent uses modify 'env'
+      env:
+        - OS_TYPE=centos OS_VERSION=6 BUILD_ENV=uw_build
+      stage: Test
+      sudo: required
+      services:
+        - docker
+      before_install:
+        - sudo apt-get update
+        - echo 'DOCKER_OPTS="-H tcp://127.0.0.1:2375 -H unix:///var/run/docker.sock -s devicemapper"' | sudo tee /etc/default/docker > /dev/null
+        - sudo service docker restart
+        - sleep 5
+        - sudo docker pull centos:centos${OS_VERSION}
+      script:
+        - tests/setup_tests.sh
+    - <<: *run_tests
+      env:
+        - OS_TYPE=centos OS_VERSION=7 BUILD_ENV=uw_build
+    - <<: *run_tests
+      env:
+        - OS_TYPE=centos OS_VERSION=6 BUILD_ENV=osg
+    - <<: *run_tests
+      env:
+        - OS_TYPE=centos OS_VERSION=7 BUILD_ENV=osg
 
+    - &deploy_rpm
+      # Template; subsequent uses modify 'env'
+      env:
+        - OS_TYPE=centos OS_VERSION=6 BUILD_ENV=uw_build
+      on:
+        repo: htcondor/htcondor-ce
```

Looks like the deploy tasks still ran. Can't we combine `on` directives like so?

```suggestion
        repo: htcondor/htcondor-ce
        tags: true
        branch: master
```

---

Last time I tried it, `tags: true` and `branch: master` didn't go together: I tried making a tag and it didn't do the deploy because 'v3.2.2.5matyas is not master'. I can add `tags: true` and make it only deploy on tags.

---

So as it stands now, we'll get RPMs on every commit to the repo? Is that what @timtheisen wants?

---

?
No, that's not how it works.

---

At least, RPMs won't be created on PRs until they're merged. If you want me to only do it on tags, I can do that too.